home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollApiImpl.java

```diff
@@ -52,6 +60,15 @@
     @Autowired
     private PollQuestionService pollQuestionService;
 
+    @Autowired
+    private PollService pollService;
+
+    @Autowired
+    private VoteService voteService;
+
+    @Autowired
+    private HomeUserDetailsService userDetailsService;
```

Comment:
Generally, I mostly think that the service layer should know about the logged user by itself, and the presentation layer has only to implement the logic for injecting logged user taken from the request. In this case, we will have independent core logic that will be ready for use in any other type of presentation layer.

But it requires revisiting our security implementation. Please create a corresponding ticket in our backlog.

Reply:
Done. I moved the getting of the principal from the security context to the service class. There I do not use HomeUserDetailsService